runtime/lua/vim/fs.lua

```diff
@@ -397,13 +495,60 @@ function M.normalize(path, opts)
     path = path:gsub('/+', '/')
   end
 
-  -- Ensure last slash is not truncated from root drive on Windows
-  if iswin and path:match('^%w:/$') then
-    return path
+  local prefix = ''
+
+  -- Split Windows paths into prefix and body to make processing easier.
+  if iswin then
+    prefix, path = split_windows_path(path)
+  -- For POSIX paths, if there are leading double slashes, set prefix to a single slash and remove a
+  -- slash from the start of the path to make the path easier to work with.
+  elseif vim.startswith(path, '//') then
+    prefix = '/'
+    path = path:sub(2)
+  end
+
+  local is_path_absolute = vim.startswith(path, '/')
+  -- Remove leading and trailing slashes from the path so split works correctly.
+  path = path:gsub('^/', ''):gsub('/$', '')
+
+  -- Split the path into components and process them.
+  local path_components = vim.split(path, '/')
+  local new_path_components = {}
+
+  for _, component in ipairs(path_components) do
```

Comment:
This block looks like it should go in a function, otherwise `normalize` is a 100 LOC which is too big IMO.

Reply:
Yeah that's fair, I'll move it to a helper function like `split_windows_path`